src/core/overflowsafe_type.hpp

```diff
@@ -138,74 +136,81 @@ class OverflowSafeInt
+template <class T> inline constexpr OverflowSafeInt<T> operator + (const int   a, const OverflowSafeInt<T> b) { return b + a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator - (const int   a, const OverflowSafeInt<T> b) { return -b + a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator * (const int   a, const OverflowSafeInt<T> b) { return b * a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator / (const int   a, const OverflowSafeInt<T> b) { return (OverflowSafeInt<T>)a / (int)b; }
 
 /* Sometimes we got uint operator OverflowSafeInt instead of vice versa. Handle that properly */
-template <class T> inline OverflowSafeInt<T> operator + (uint  a, OverflowSafeInt<T> b) { return b + a; }
-template <class T> inline OverflowSafeInt<T> operator - (uint  a, OverflowSafeInt<T> b) { return -b + a; }
-template <class T> inline OverflowSafeInt<T> operator * (uint  a, OverflowSafeInt<T> b) { return b * a; }
-template <class T> inline OverflowSafeInt<T> operator / (uint  a, OverflowSafeInt<T> b) { return (OverflowSafeInt<T>)a / (int)b; }
+template <class T> inline constexpr OverflowSafeInt<T> operator + (const uint  a, const OverflowSafeInt<T> b) { return b + a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator - (const uint  a, const OverflowSafeInt<T> b) { return -b + a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator * (const uint  a, const OverflowSafeInt<T> b) { return b * a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator / (const uint  a, const OverflowSafeInt<T> b) { return (OverflowSafeInt<T>)a / (int)b; }
 
 /* Sometimes we got byte operator OverflowSafeInt instead of vice versa. Handle that properly */
-template <class T> inline OverflowSafeInt<T> operator + (byte  a, OverflowSafeInt<T> b) { return b + (uint)a; }
-template <class T> inline OverflowSafeInt<T> operator - (byte  a, OverflowSafeInt<T> b) { return -b + (uint)a; }
-template <class T> inline OverflowSafeInt<T> operator * (byte  a, OverflowSafeInt<T> b) { return b * (uint)a; }
-template <class T> inline OverflowSafeInt<T> operator / (byte  a, OverflowSafeInt<T> b) { return (OverflowSafeInt<T>)a / (int)b; }
+template <class T> inline constexpr OverflowSafeInt<T> operator + (const byte  a, const OverflowSafeInt<T> b) { return b + (uint)a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator - (const byte  a, const OverflowSafeInt<T> b) { return -b + (uint)a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator * (const byte  a, const OverflowSafeInt<T> b) { return b * (uint)a; }
+template <class T> inline constexpr OverflowSafeInt<T> operator / (const byte  a, const OverflowSafeInt<T> b) { return (OverflowSafeInt<T>)a / (int)b; }
 
 typedef OverflowSafeInt<int64> OverflowSafeInt64;
 typedef OverflowSafeInt<int32> OverflowSafeInt32;
 
+/* Some basic "unit tests". Also has the bonus of confirming that constexpr is working */
```

Comment:
If we are nitpicking anyway, mind adding dots at the end of sentences?

Reply:
I was following the convention of the current file :p